### ISSUE TITLE
Add Dark Mode button

### ISSUE THREAD

**Issue by Devnol** [hackropolis/website#11]
The auto-change of light and dark mode on the website is really cool, but some people like me like having the system auto-switch while some apps like Slack, Discord or some website change their theme independently. I was thinking of adding a button at the bottom corner of the website for switching the theme. That sounds easy enough and I'll try to do as much as I can myself but any help would be appreciated. (I'd like to learn something in the process so if someone else is gonna do it at least explain how and why it works).
I've already made a branch  so before you edit anything make sure to checkout to themeswitcher.

**Comment by GeopJr** [comment]
There are 3 parts to this:
- Changing the colors
- Reading/Writing user preference
- Dealing with logos like Vercel's which right now uses the `Picture` element and depends on prefers-color-scheme

### Changing the colors
The best way to do this is probably by splitting the root variables into the light theme ones and a class for the dark theme ones. (https://github.com/hackropolis/website/blob/main/styles/globals.css#L17-L22)
Something like this would probably do:
```css
:root {
	--text: #000000;
	--bg: #ffffff;
}

[data-theme="dark"] {
	--text: #ffffff;
	--bg: #000000;
}
```

Now any element that has data-theme="dark" will get the dark variables, like `<div data-theme="dark" />`.

### Reading/Writing user preference
This will use JS' localStorage#getItem and localStorage#setItem. After reading which theme the user prefers, you then add the `data-theme="dark"` (or remove it in case of light theme) to `Document#documentElement`. That should do it.

By default, since the user might not have chosen a theme yet, you need to check using JS' their `prefers-color-scheme` and do the same function of adding or removing the attribute. `window#matchMedia` will allow you to check the prefers-color-scheme they match.
Don't forget to set the localStorage entry to that.

### Dealing with logos like Vercel's which right now uses the `Picture` element and depends on prefers-color-scheme
React hooks lol

**Comment by cfanoulis** [comment]
the prop should be named data-theme-color to conform to html standard thanks

**Comment by Devnol** [comment]
Thanks for the Insight Geop, that's pretty helpful.
Charalampos sure whatever works for you ig.

The way I was thinking we made this was use font-awesome's Sun and Moon SVGs, have a button that when hovered upon says "switch to light" or "switch to dark" (see [bc395be](https://github.com/hackropolis/website/commit/bc395be27a9d16b3692b79c386c9c8e5ef917e01)) and when clicked change that and the displayed icon. Doesn't sound that hard and I found some info [here](https://stackoverflow.com/questions/57762858/how-can-i-use-floating-button-in-my-react-web-site) I think.
